### PR TITLE
Appstream content

### PR DESCRIPTION
**Pull request by Loquacity**
https://github.com/SUSE/spacewalk/issues/10705

(follows on from PR #129)

### CODE REVIEW

**Comment by cbbayburt** [comment]
I think it's better if I prepare a draft and we start from there.

@Loquacity IIUC, the preferred structure is to mention the feature briefly in the RH client doc, then describe the details in a separate section in the CLM doc. Right?

**Comment by Loquacity** [comment]
> I think it's better if I prepare a draft and we start from there.
> 
> @Loquacity IIUC, the preferred structure is to mention the feature briefly in the RH client doc, then describe the details in a separate section in the CLM doc. Right?

OK, that would probably be good. I admit, I'm struggling to understand this. I don't have a preference at this point, once you've got a draft I can rearrange things into the right place, though. Thanks Can!

**Comment by cbbayburt** [comment]
@Loquacity sent the draft via email

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 1]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+
+
+
+.Procedure: Using Appstream Filters
+. In the {productname} {webui}, navigate to your {rhel}{nbsp}8 CLM project.
+Ensure that you have included the Appstream channels for your project.
+. Click btn:``Create Filter`` and use these parameters:
++
+* In the [guimenu]``Filter Name`` field, type a name for the new filter.
+* In the [guimenu]``Filter Type`` field, select [parameter]``Module (Stream)``
```
Comment (keichwa) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
* In the [guimenu]``Filter Type`` field, select [parameter]``Module (Stream)``.
```
:tada: ;)

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 2]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+* In the [guimenu]``Module Name`` field, type a module name.
+For example, [parameter]``postgresql``.
+* In the [guimenu]``Stream`` field, type the name of the desired stream.
+For example; [parameter]``10``.
```
Comment (keichwa) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
For example, [parameter]``10``.
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 3]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+If you leave this field blank, the default stream for the module is selected.
+. Click btn:[Save] to create the new filter.
+. Navigate to menu:Content Lifecycle[Projects] and select your project.
+. Click btn:``Attach/Detach Filters``, select your new Appstream filter, and click btn:``Save``.
```
Comment (keichwa) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
. Click btn:``Attach/Detach Filters``, select your new Appstream filter, and click btn:[Save].
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 4]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+
+[[NOTE]]
+====
+Be careful not to specify conflicting, incompatible, or missing
+module streams.
+For example, selecting two streams from the same module is
+invalid.
+====
+
+When you build your CLM project using the btn:``Build`` button in teh {webui}, the target repository is a regular repository without any modules, that contains packages from the selected module streams.
```
Comment (keichwa) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
When you build your CLM project using the btn:[Build] button in the {webui}, the target repository is a regular repository without any modules, that contains packages from the selected module streams.
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 5]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
 The new kernel will remain valid for one year.
 All packages installed during the year will match the current live patching kernel filter.
+
+
+
+== Appstream Filters
+
+If you are using {rhel}{nbsp}8 clients, you cannot perform package operations such as installing or upgrading directly from modular repositories like the {rhel} Appstream repository.
+You can use the Appstream filter to transform modular repositories into regular repositories.
+It does this by keeping the the packages in the repository and stripping away the module metadata.
```
Comment (cbbayburt) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
It does this by keeping the packages in the repository and stripping away the module metadata.
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 6]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+The resulting repository can be used in {productname} in the same way as a regular repository.
+
+The AppStream filter selects a single module stream to be included in the target repository.
+You can add multiple filters to select multiple module streams.
+
+If you do not use an AppStream filter in your CLM project, the module metadata in the modular sources remains intact, and the target repositories contain the same module metadata.
+As long as at least one AppStream filter is enabled in the CLM project, all target repositories are transformed into regular repositories.
+
+To use the Appstream filter, you need a CLM project with a modular repository such as ``{rhel} Appstream``.
```
Comment (cbbayburt) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
To use the AppStream filter, you need a CLM project with a modular repository such as ``{rhel} AppStream``.
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 7]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+Ensure that you included the module you need as a source before you begin.
+
+
+
+.Procedure: Using Appstream Filters
+. In the {productname} {webui}, navigate to your {rhel}{nbsp}8 CLM project.
+Ensure that you have included the Appstream channels for your project.
```
Comment (cbbayburt) [inline modules/administration/pages/content-lifecycle-examples.adoc]:
```suggestion
Ensure that you have included the AppStream channels for your project.
```

File: modules/administration/pages/content-lifecycle-examples.adoc [hunk 8]
```diff
@@ -258,3 +217,60 @@ Systems using these channels will have the kernel update available for installat
...
+====
+
+When you build your CLM project using the btn:[Build] button in the {webui}, the target repository is a regular repository without any modules, that contains packages from the selected module streams.
+
+// This feature is mandatory to make RHEL/CentOS 8 modular repositories work
+// in the SUMA UI. Otherwise, even though modular repositories can be synced
+// and assigned to clients, they cannot be used for package operations
+// (install, update, etc.) from SUMA UI (An info message is shown in the UI if
+// that's the case). In that case, the package operations can only be done
+// from the client's CLI manually.
```
Comment (cbbayburt) [inline modules/administration/pages/content-lifecycle-examples.adoc]: This paragraph was meant for you to read :smile: 
But I guess it makes sense to keep it in as a comment.

Comment (Loquacity) [reply modules/administration/pages/content-lifecycle-examples.adoc]: Eh, now it's for all of us to read :wink:

**Pull request merged**